### DISCUSSION TITLE
Unable to push the code using git push

### DISCUSSION THREAD

**Discussion by ghost** [community/community#56994] [Repositories]
### Select Topic Area

General

### Select Feature Area

- [X] Issues
- [ ] Projects

### Body

I have remote repository, but when I tried to push the code using `git push -u` I get an error 

`ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.`

I have previously uninstall and install the window after that I have generate the new ssh key but now I'm facing that issue.

Please help 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by nirav-gajera** [comment]
Verify the repository URL by running the following command in your local repository:
`git remote -v`

Check your SSH key setup by running the following command to test your SSH connection to the remote repository:
`ssh -T git@<repository-url>
`
Replace <repository-url> with the actual URL of your remote repository. 

try git pull before push

**Reply by ghost** [reply]
I run this same command in my local repository `git remote -v` and I get URL `git@github.com:AliFarooq-dev/Cloud_Knox.git` for push and fetch.

But when I try to check SSH key using above command I get error of `Could not resolve hostname github.com:AliFarooq-dev/Cloud_Knox.git: No such host is known`.

Also try git pull but it give error :

`ERROR: Repository not found.
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.`

What should I do ?

**Reply by airtower-luna** [reply]
> But when I try to check SSH key using above command

That command is wrong, correct would be `ssh -vT git@github.com`. The `-v` means you'll get debug output (which you'll need to figure out what's wrong if it doesn't work), and if you include the whole repository URL it'll be treated as the hostname, leading to the error you're seeing.

**Reply by ghost** [reply]
Thanks @airtower-luna !

I have try this command `ssh -vT git@github.com` but now I'm seeing this error in the last of debug `Hi AliFarooq-dev! You've successfully authenticated, but GitHub does not provide shell access`.

Git Hub does not provide me access I don't know why. I try to figure out but still confuse.

**Reply by ghost** [reply]
full debug is:

`debug1: Connection established.
debug1: identity file C:\\Users\\DELL/.ssh/id_rsa type -1
debug1: identity file C:\\Users\\DELL/.ssh/id_rsa-cert type -1
debug1: identity file C:\\Users\\DELL/.ssh/id_dsa type -1
debug1: identity file C:\\Users\\DELL/.ssh/id_dsa-cert type -1
debug1: identity file C:\\Users\\DELL/.ssh/id_ecdsa type -1
debug1: identity file C:\\Users\\DELL/.ssh/id_ecdsa-cert type -1
debug1: identity file C:\\Users\\DELL/.ssh/id_ed25519 type 3
debug1: identity file C:\\Users\\DELL/.ssh/id_ed25519-cert type -1
debug1: identity file C:\\Users\\DELL/.ssh/id_xmss type -1
debug1: identity file C:\\Users\\DELL/.ssh/id_xmss-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_for_Windows_8.1
debug1: Remote protocol version 2.0, remote software version babeld-51390fc7
debug1: no match: babeld-51390fc7
debug1: Authenticating to github.com:22 as 'git'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256
debug1: kex: host key algorithm: ssh-ed25519
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ssh-ed25519 SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU
debug1: Host 'github.com' is known and matches the ED25519 host key.
debug1: Found key in C:\\Users\\DELL/.ssh/known_hosts:1
debug1: rekey out after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey in after 134217728 blocks
debug1: pubkey_prepare: ssh_get_authentication_socket: No such file or directory
debug1: Will attempt key: C:\\Users\\DELL/.ssh/id_rsa
debug1: Will attempt key: C:\\Users\\DELL/.ssh/id_dsa 
debug1: Will attempt key: C:\\Users\\DELL/.ssh/id_ecdsa
debug1: Will attempt key: C:\\Users\\DELL/.ssh/id_ed25519 ED25519 SHA256:4Jenhio4rsFfCX282PDiWCG8LI/RqV4alE3BtyRxppY
debug1: Will attempt key: C:\\Users\\DELL/.ssh/id_xmss 
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp256-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,rsa-sha2-512-cert-v01@openssh.com,rsa-sha2-256-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,sk-ssh-ed25519@openssh.com,sk-ecdsa-sha2-nistp256@openssh.com,ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,rsa-sha2-512,rsa-sha2-256,ssh-rsa>    
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey
debug1: Next authentication method: publickey
debug1: Trying private key: C:\\Users\\DELL/.ssh/id_rsa
debug1: Trying private key: C:\\Users\\DELL/.ssh/id_dsa
debug1: Trying private key: C:\\Users\\DELL/.ssh/id_ecdsa
debug1: Offering public key: C:\\Users\\DELL/.ssh/id_ed25519 ED25519 SHA256:4Jenhio4rsFfCX282PDiWCG8LI/RqV4alE3BtyRxppY
debug1: Server accepts key: C:\\Users\\DELL/.ssh/id_ed25519 ED25519 SHA256:4Jenhio4rsFfCX282PDiWCG8LI/RqV4alE3BtyRxppY
debug1: Authentication succeeded (publickey).
Authenticated to github.com ([20.207.73.82]:22).
debug1: channel 0: new [client-session]
debug1: Entering interactive session.
debug1: Authentication succeeded (publickey).
Authenticated to github.com ([20.207.73.82]:22).
debug1: channel 0: new [client-session]
debug1: Entering interactive session.
debug1: Authentication succeeded (publickey).
Authenticated to github.com ([20.207.73.82]:22).
debug1: channel 0: new [client-session]
debug1: Entering interactive session.
debug1: pledge: network
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
Hi AliFarooq-dev! You've successfully authenticated, but GitHub does not provide shell access.
debug1: channel 0: free: client-session, nchannels 1`

**Reply by airtower-luna** [reply]
That message means your SSH key is set up correctly, but somehow doesn't work (or isn't used) when you try to use it via Git. I don't use Windows myself, but I know a lot of people using it have trouble with Git Bash and Powershell disagreeing where the user's home directory is, and thus the SSH keys. Try doing your Git operations from Git Bash if you've been using Powershell (or worse, CMD) so far.

For future reference, that log would be a lot more readable in a [code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks). :wink:

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @AliFarooq-dev ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
